zkevm-circuits/src/evm_circuit/execution/returndatacopy.rs

```diff
@@ -64,18 +60,27 @@ impl<F: Field> ExecutionGadget<F> for ReturnDataCopyGadget<F> {
         let opcode = cb.query_cell();
 
         let dest_offset = cb.query_cell_phase2();
-        let data_offset = cb.query_word_rlc();
-        let size = cb.query_word_rlc();
-
+        let return_data_size: Cell<F> = cb.query_cell();
+
+        let size: RandomLinearCombination<F, N_BYTES_MEMORY_ADDRESS> = cb.query_word_rlc();
```

Comment:
(optional)

The `size` variable can be replaced with `check_overflow_gadget.size()`:

    let size = RandomLinearCombination::new(check_overflow_gadget.size().cells[..N_BYTES_MEMORY_ADDRESS].clone(), cb.challenges.evm_word());

And instead of this `require_equal`, constrain the remaining bytes to be 0:

    cb.require_zero(sum::expr(&check_overflow_gadget.size().cells[N_BYTES_MEMORY_ADDRESS..]));